### PR TITLE
[CRT] Add _Exit

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Add _Exit to process.h

This is already done this way in our stdlib.h. It is needed by GCC 13 C++ headers. The problem happens like this:
- telnet/precomp.h includes fstream from GCC
- fstream includes pthread.h from GCC
- pthread.h includes process.h from ReactOS
- process.h defines _CRT_TERMINATE_DEFINED, but doesn't declare _Exit
- fstream includes cstdlib from GCC
- cstdlib includes stdlib.h from GCC (#include_next)
- stdlib.h doesn't declare _Exit, because _CRT_TERMINATE_DEFINED is defined
- cstdlib uses _Exit

### CODE REVIEW

**Review by HBelusca — APPROVED** [review]
Since this is GCC workaround, it would be of interest to surround that with an `#ifdef __GNUC__`

**Review by cbialorucki — APPROVED** [review]
Please answer Hermes' question and guard this with `#ifdef __GNUC__` since this is a GCC-specific workaround.

**Comment by tkreuzer** [comment]
> Since this is GCC workaround, it would be of interest to surround that with an `#ifdef __GNUC__`
> 
> Question (since it's not fully clear): Is this happening also in C projects? or C++ only? I'm asking because in the example scenario you say:
> 
> > #include GCC _**c++ header**_

It turns out it isn't actually a workaround as I though. It's a a C99 definition and MS headers have it, too, just not inline, because ucrtbase exports it.
I updated the commit message with some more details of what is going on when compiling c++ code with GCC.

File: sdk/include/crt/process.h [hunk 1]
```diff
@@ -54,15 +54,19 @@ extern "C" {
 #define _CRT_TERMINATE_DEFINED
   __declspec(noreturn) void __cdecl exit(_In_ int _Code);
   _CRTIMP __declspec(noreturn) void __cdecl _exit(_In_ int _Code);
-
+#if !defined __NO_ISOCEXT /* extern stub in static libmingwex.a */
+  /* C99 function name */
+  __declspec(noreturn) void __cdecl _Exit(int); /* Declare to get noreturn attribute.  */
```
Comment (HBelusca) [inline sdk/include/crt/process.h:59]:
```suggestion
  __declspec(noreturn) void __cdecl _Exit(int); /* Declare to get noreturn attribute */
```

**Review by HBelusca — APPROVED** [review]

**Review by cbialorucki — APPROVED** [review]

**Pull request merged**